deft.el

```diff
@@ -981,7 +986,29 @@ See `deft-find-all-files'."
                  (not (backup-file-name-p file))
                  (member (file-name-extension file) deft-extensions))
             (setq result (cons file result)))))
-        result)))
+        (deft-apply-generation-rules result))))
+
+(defun deft-apply-generation-rules (lst)
+  (let ((result nil))
+    (when deft-generation-rules
+    (dolist (file lst)
+      (when (not (deft-exported-from-previous? file deft-generation-rules result))
```

Comment:
Shouldn't we pas `lst' to `deft-exported-from-previous?' instead of `result'. `result' is a partially complete list of files, which is being updated in the loop, while `lst' is the entire list. Otherwise, I think the result will depend on the order in which files appear in the list (i.e., whether the source file has already been processed when looking at the generated file).